File: striver_series/src/SubarrayWithSum.java

public class SubarrayWithSum {

    public static void main(String[] args) {
        int[] arr = { -1, 1, 1};
        int k = 1;
        int maxSubArrayLen = greatestSubArrayWithSum(arr,k);
        System.out.println("Max subArray with given sum: "+maxSubArrayLen);
    }

    static int greatestSubArrayWithSum(int [] arr,int k){
        int len  = 0;
       for(int i=0;i<arr.length;i++){
           int sum = 0;
           for(int j=i;j<arr.length;j++){
               sum +=arr[j];
               if(sum==k){
                   len = Math.max(len,j-i+1);
               }
           }
       }
        return len;
    }
}

File: dart_programs/practise_programs/sub_array_with_sum.dart
import 'dart:math';

void main() {
  List<int> arr = [-1, 1, 1];
  int k = 1;
  int maxSubArrayLen = greatestSubArrayWithSum(arr, k);
  print("Max subArray with given sum: $maxSubArrayLen");
}

int greatestSubArrayWithSum(List<int> arr, int k) {
  int len = 0;
  for (int i = 0; i < arr.length; i++) {
    int sum = 0;
    for (int j = i; j < arr.length; j++) {
      sum += arr[j];
      if (sum == k) {
        len = max(len, j - i + 1);
      }
    }
  }
  return len;
}
